core-js/src/test/javascript/cdf/components/UnmanagedComponent-spec.js

```diff
@@ -219,10 +222,85 @@ define([
     });
 
     /**
+     * ## Unmanaged Component # should allow changes in the datasources
+     */
+    it("should allow changes in the datasources", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition.dataSource = "otherDatasource";
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+    });
+
+    /**
+     * ## Unmanaged Component # should allow changes in the datasources by override
+     */
+    it("should allow changes  in the datasources by override", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition = $.extend( {}, this.queryDefinition , { dataSource: "otherDatasource"});
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+     });
+
+    /**
+     * ## Unmanaged Component # should allow changes in the datasources by merge
+     */
+    it("should allow changes datasources by  merge", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition = $.extend( this.queryDefinition , { dataSource: "otherDatasource"});
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+    });
+
+    /**
+     * ## Unmanaged Component # should recognize a function as a datasource if defined directly
+     */
+    it("should recognize a function as a datasource if defined directly", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      getQueryDef = function () { return { dataSource: "otherDatasource"}; };
+      uquery.preExecution = function () {
+        uquery = $.extend( uquery , {getQueryDefinition : getQueryDef });
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+     });
+
+    /**
+     * ## Unmanaged Component # should recognize a function as a datasource if defined in a property
+     */
+    it(" should recognize a function as a datasource if defined in a property", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        uquery.getQueryDefinition = function () { return { dataSource: "otherDatasource"}; };
```

Comment:
Isn't this test equal to the one above? The only difference being that the previous test uses `$.extend` to set the `getQueryDefinition` property.